pipeline/mutate/mutator_hydrator_test.go

```diff
@@ -346,6 +353,15 @@ func TestMutatorHydrator(t *testing.T) {
 				Match:   newAuthenticationSession(),
 				Err:     nil,
 			},
+			"Custom Cache Key": {
+				Setup:   defaultRouterSetup(),
+				Session: newAuthenticationSession(setSubject(sampleSubject)),
+				Rule:    &rule.Rule{ID: "test-rule"},
+				Config:  configWithSpecialCacheKey(sampleSubject),
```

Comment:
According to the documentation, this should work with substitution / templating. It would be good if you could add some tests that cover this, and also ensure that no cache key is the same when templating is executed appropriately.

Reply:
Done.